### DISCUSSION TITLE
check agent availability before starting a conversation

### DISCUSSION THREAD

**Discussion by EdwinWortel** [chatwoot/chatwoot#2808] [Q&A]
Hi there, I have a question;
I am building a chatbot, and at a certain point when the bot cannot heIp anymore, I want to handover the user to a custom inbox to an available human agent.
Now, I only want to do that when a human agent really is available to take that live chat. If no human agent is available, I keep the user in the bot and follow another path.
Would you have an API access to see beforehand if an human agent is available?

So I need at least 1 agent who is online/available/green before handing over.

**Comment by sojan-official** [comment]
We have the API to list agents which also contains the online status of the agents. I think the documentation is missing for this API, we will add it to our swagger.

```
curl --location --request GET 'http://localhost:3000/api/v1/accounts/1/agents' \
--header 'api_access_token: isnjJH2WTcD1oziBU4CkGsoJ'
```
